catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java

```diff
@@ -836,9 +858,126 @@ public Response addDocument(@Context HttpHeaders headers, @Context UriInfo reque
         return response;
     }
 
+    CreateInfo parseAttachments(List<Attachment> contentParts, String transformerParam) {
+
+        if (contentParts.size() == 1) {
+            Attachment contentPart = contentParts.get(0);
+            return parseAttachment(contentPart);
+        }
+        List<Attribute> attributes = new ArrayList<>(contentParts.size());
+        Metacard metacard = null;
+        CreateInfo createInfo = null;
+
+        for (Attachment attachment : contentParts) {
+            String name = attachment.getContentDisposition()
+                    .getParameter("name");
+            String parsedName = (name.startsWith("parse.")) ? name.substring(6) : name;
+            try {
+                InputStream inputStream = attachment.getDataHandler()
+                        .getInputStream();
+                if (name.equals("parse.metadata")) {
+                    String transformer = DEFAULT_METACARD_TRANSFORMER;
+                    if (transformerParam != null) {
+                        transformer = transformerParam;
+                    }
+                    try {
+                        MimeType mimeType = new MimeType(attachment.getContentType()
+                                .toString());
+                        metacard = generateMetacard(mimeType,
+                                "assigned-when-ingested", inputStream, transformer);
+                    } catch (MimeTypeParseException | MetacardCreationException e) {
+                        LOGGER.debug("Unable to parse metadata {}",
+                                attachment.getContentType()
+                                        .toString());
+                    } finally {
+                        IOUtils.closeQuietly(inputStream);
+                    }
+
+                } else if (name.equals("parse.resource")) {
```

Comment:
If you switched the order of these first two tests, you could defer the retrieval of the inputStream a bit. Not a huge improvement, but it would be a bit cleaner and more clear.